### PR TITLE
Deposit form complies with WCAG A/AA accessibility levels

### LINKED ISSUE #3149: Comply with WCAG A/AA accessibility levels: deposit form
See https://docs.google.com/document/d/1M2tzjyHLzE8oRXLS3KAJ2n_GO9l-0QKq/edit?rtpof=true

### PR DESCRIPTION
**Pull request by ndushay**
# Why was this change made? 🤔

Closes #3149

Addressed all A and AA accessibility problems that I could find with SiteImprove, WAVE and Axe plugins.  Note that Axe found no violations in the original.

# Before

image

image

# After 

image

- ^^ note that the remaining all need design help

image

- ^^ I was unable to figure out the hidden fields causing the 4 errors still indicated by Wave, so I'm declaring this PR an excellent first cut.

# How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

# Does your change introduce accessibility violations? 🩺

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

File: app/components/works/add_files_component.html.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@
       <div data-controller="dropzone" data-dropzone-max-file-size="10000" data-dropzone-previews-container=".dropzone-files-previews">
         <fieldset>
           <div class="dropzone dropzone-default" data-dropzone-target="container">
-            <%= form.file_field :files, multiple: true, direct_upload: true,
+            <%= form.file_field :files, multiple: true, direct_upload: true, namespace: "flat_list", "aria-label": "Upload files",
```
Comment (ndushay) [inline app/components/works/add_files_component.html.erb:18]: adding a namespace makes the ids generated unique compared with other namespaces ...

File: app/components/works/contributor_row_component.html.erb [hunk 2]
```diff
@@ -39,23 +40,24 @@
             <%= form.label :with_orcid, "Enter ORCID iD", value: "true", class: "form-check-label fw-semibold" %>
           </div>
           <div class="row person-identifier-row" data-contributors-target="personOrcid">
+            <% text_field_id_prefix = "work_#{namespace}_orcid_#{form.options[:child_index]}_" %>
             <div class="col">
-              <%= form.label :orcid, orcid_label, class: "form-label" %>
+              <%= form.label :orcid, orcid_label, for: "#{text_field_id_prefix}id", class: "form-label" %>
```
Comment (ndushay) [inline app/components/works/contributor_row_component.html.erb:45]: "for" hooks binds this label to the text field, and the expression ensures that each contributor has unique labels.

File: app/components/works/keywords_row_component.html.erb [hunk 3]
```diff
@@ -1,14 +1,15 @@
 <div class="mb-3 row plain-container g-0 keyword-row">
   <div class="col-sm-6">
-    <%= form.label :label, "Keyword", class: "col-form-label visually-hidden" %>
-    <div data-controller="keywords autocomplete autocomplete-edit" data-autocomplete-url-value="/autocomplete" data-autocomplete-min-length-value="3" class="dropdown" role="combobox">
-      <%= form.text_field :label, {class: "form-control#{"is-invalid" if error?}", required: true, "data-autocomplete-target": "input", "data-autocomplete-edit-target": "input",
+    <% unique_field_id = "work_keywords_attributes_#{form.options[:child_index]}" %>
+    <div data-controller="keywords autocomplete autocomplete-edit" data-autocomplete-url-value="/autocomplete" data-autocomplete-min-length-value="3" aria-label="autocomplete group" class="dropdown" role="combobox" aria-expanded="false">
```
Comment (ndushay) [inline app/components/works/keywords_row_component.html.erb:4]: it seemed weird that there were complaints about a div not having a label, but ... doing this got rid of the error (siteImprove, as I recall)

File: app/components/works/keywords_row_component.html.erb [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
+      <%= form.label :label, "Keyword", for: unique_field_id, class: "col-form-label visually-hidden" %>
```
Comment (ndushay) [inline app/components/works/keywords_row_component.html.erb:5]: moved this label to live with the text field it is for.

File: app/javascript/controllers/dropzone_controller.js [hunk 5]
```diff
@@ -240,15 +240,17 @@ class DirectUploadController {
     input.type = "hidden";
     input.classList.add('hidden-file')
     input.name = `work[attached_files_attributes][${this.count}][file]`
+    input.setAttribute("aria-label", "hidden file");
```
Comment (ndushay) [inline app/javascript/controllers/dropzone_controller.js:243]: actual change: adding aria-label

File: app/javascript/controllers/dropzone_controller.js [hunk 6]
```diff
@@ -240,15 +240,17 @@ class DirectUploadController {
...
     return input;
   }
 
   createHiddenPathInput() {
     const input = document.createElement("input")
     input.type = "hidden"
-    input.name = `work[attached_files_attributes][${this.count}][path]`    
+    input.name = `work[attached_files_attributes][${this.count}][path]`
     input.value = this.source.dirPath ? `${this.source.dirPath}/${this.file.name}` : this.file.name
+    input.setAttribute("aria-label", "hidden path");
```
Comment (ndushay) [inline app/javascript/controllers/dropzone_controller.js:253]: actual change: adding aria-label.

**Review by mjgiarlo — APPROVED** [review]
I assume if the form field namespace or ID changes were going to break anything, we'd be seeing broken tests? If that assumption is true, :shipit: If not, maybe worth running the integration tests for H2?

**Pull request merged**